backend/global_knowledge_collector.py

```
@@ -156,30 +156,36 @@ async def collect_rss_feed(self, rss_url: str, category: str, max_articles: int
             # Process entries
             for entry in feed.entries[:max_articles]:
                 try:
-                    # Clean and prepare content
-                    title = self.clean_content(entry.title)
+                    # Clean and prepare content - using getattr for safety
+                    title = getattr(entry, "title", "")
+                    link = getattr(entry, "link", None) or url  # fallback to feed URL
```

Ruff (0.12.2)

Undefined name `url`

(F821)